bypros/test/integration_test.go

```diff
@@ -0,0 +1,488 @@
+package bypros
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/cothority/v3/bypros"
+	"go.dedis.ch/cothority/v3/bypros/storage/sqlstore"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/contracts"
+)
+
+func TestIntegration_Simple(t *testing.T) {
+	os.Setenv("PROXY_DB_URL", "postgres://bypros:docker@localhost:4567/bypros")
```

Comment:
These test functions are huge and there's no way I can spot a bug in something this big. 

I'm not sure how to approach this though, as I imagine it's not necessarily an easy refactoring?

Reply:
We could refactor the test functions to use multiple sub-functions, but I'm afraid we would loose in readability. IMO integration tests should offer a single "thread of events" that one can follow to see how the system work in an "end-to-end" fashion.